src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/PackIntegrationTest.cs

```diff
@@ -176,5 +177,126 @@ public async Task Pack_IncludesRazorFilesAsContent_WhenIncludeRazorContentInPack
                 Path.Combine("bin", Configuration, "ClassLibrary.1.0.0.nupkg"),
                 Path.Combine("lib", "netcoreapp3.0", "ClassLibrary.Views.dll"));
         }
+
+        [Fact]
+        [InitializeTestProject("PackageLibraryDirectDependency", additionalProjects: new[] { "PackageLibraryTransitiveDependency" })]
+        public async Task Pack_IncludesStaticWebAssets()
+        {
+            var result = await DotnetMSBuild("Pack");
+
+            Assert.BuildPassed(result, allowWarnings: true);
+
+            Assert.FileExists(result, OutputPath, "PackageLibraryDirectDependency.dll");
+
+            Assert.NupkgContains(
+                result,
+                Path.Combine("..", "TestPackageRestoreSource", "PackageLibraryDirectDependency.1.0.0.nupkg"),
+                filePaths: new[]
+                {
+                    Path.Combine("staticwebassets", "js", "pkg-direct-dep.js"),
+                    Path.Combine("staticwebassets", "css", "site.css"),
+                    Path.Combine("build", "Microsoft.AspNetCore.StaticWebAssets.props"),
+                    Path.Combine("build", "PackageLibraryDirectDependency.props"),
+                    Path.Combine("buildMultiTargeting", "PackageLibraryDirectDependency.props"),
+                    Path.Combine("buildTransitive", "PackageLibraryDirectDependency.props")
+                });
+        }
+
+        [Fact]
+        [InitializeTestProject("PackageLibraryDirectDependency", additionalProjects: new[] { "PackageLibraryTransitiveDependency" })]
+        public async Task Pack_NoBuild_IncludesStaticWebAssets()
+        {
+            var result = await DotnetMSBuild("Build");
+            Assert.BuildPassed(result, allowWarnings: true);
+
+            var pack = await DotnetMSBuild("Pack", "/p:NoBuild=true");
+            Assert.BuildPassed(pack, allowWarnings: true);
+
+            Assert.FileExists(pack, OutputPath, "PackageLibraryDirectDependency.dll");
+
+            Assert.NupkgContains(
+                pack,
+                Path.Combine("..", "TestPackageRestoreSource", "PackageLibraryDirectDependency.1.0.0.nupkg"),
+                filePaths: new[]
+                {
+                    Path.Combine("staticwebassets", "js", "pkg-direct-dep.js"),
+                    Path.Combine("staticwebassets", "css", "site.css"),
+                    Path.Combine("build", "Microsoft.AspNetCore.StaticWebAssets.props"),
+                    Path.Combine("build", "PackageLibraryDirectDependency.props"),
+                    Path.Combine("buildMultiTargeting", "PackageLibraryDirectDependency.props"),
+                    Path.Combine("buildTransitive", "PackageLibraryDirectDependency.props")
+                });
+        }
+
+        [Fact]
+        [InitializeTestProject("ComponentLibrary")]
+        public async Task Pack_DoesNotIncludeAnyCustomPropsFiles_WhenNoStaticAssetsAreAvailable()
```

Comment:
Can we add a test that verifies stuff doesn't get generated when the `EnableWebStaticContent` is disabled?

Reply:
Added the test